session.go

```diff
@@ -178,7 +187,18 @@ func (s *Session) startPings(rootCtx context.Context) {
 				if err := s.sendPing(); err != nil {
 					logrus.WithError(err).Error("Error writing ping")
 				}
-				logrus.Debug("Wrote ping")
+				v := ctx.Value(ContextKeyCaller)
+				s := ""
+				if v != nil {
+					s1, ok := v.(string)
```

Comment:
We should check for `s1, ok := v.(ContextKey)` here just for clarity

Reply:
Sorry, this commit is out of date.  The struct-based code does that implicitly -- if there's no entry for `ContextKeyCaller`, we get back `nil` and return empty string.  And then if for some reason we can't get the string associated with that entry, we return empty string.

Reply:
LGTM